packages/@uppy/provider-views/src/SearchProviderView/InputView.jsx

```diff
@@ -1,17 +1,27 @@
 import { h } from 'preact'
+import { useEffect } from 'preact/hooks'
+import { nanoid } from 'nanoid/non-secure'
 
 export default ({ i18n, search }) => {
   let input
-  const validateAndSearch = () => {
+  const validateAndSearch = (ev) => {
+    ev.preventDefault()
     if (input.value) {
       search(input.value)
     }
   }
-  const handleKeyPress = (ev) => {
-    if (ev.keyCode === 13) {
-      validateAndSearch()
+
+  const form = document.createElement('form')
```

Comment:
I think the form shouldn't be recreated and appended to body on every re-render. Also, I think it's preferred to use state instead of refs for handling input state in react. Maybe something like this:

```js
import { h } from 'preact'
import { useEffect, useId, useState, useCallback } from 'preact/hooks'

export default ({ i18n, search }) => {
  const [searchText, setSearchText] = useState('')

  const validateAndSearch = useCallback((ev) => {
    ev.preventDefault()
    if (searchText) search(searchText)
  }, [search, searchText])

  const formId = useId()

  const [form] = useState(() => {
    const form2 = document.createElement('form')
    form2.id = formId
    return form2
  })

  useEffect(() => {
    document.body.appendChild(form)
    return () => document.body.removeChild(form)
  }, [form])

  useEffect(() => {
    form.addEventListener('submit', validateAndSearch)
    return () => form.removeEventListener('submit', validateAndSearch)
  }, [form, validateAndSearch])

  const onSearchTextChange = useCallback((e) => setSearchText(e.target.value), [])

  return (
    <div className="uppy-SearchProvider">
      <input
        className="uppy-u-reset uppy-c-textInput uppy-SearchProvider-input"
        type="search"
        aria-label={i18n('enterTextToSearch')}
        placeholder={i18n('enterTextToSearch')}
        value={searchText}
        onInput={onSearchTextChange}
        data-uppy-super-focusable
        form={formId}
      />
      <button
        className="uppy-u-reset uppy-c-btn uppy-c-btn-primary uppy-SearchProvider-searchButton"
        type="submit"
        form={formId}
      >
        {i18n('searchImages')}
      </button>
    </div>
  )
}
```

Note that to use `useId` we would have to update root package.json `resolutions`:
```
    "preact": "patch:preact@npm:10.11.3#.yarn/patches/preact-npm-10.10.0-dd04de05e8.patch"
```